tests/console/validate_multipath.pm

```diff
@@ -82,46 +66,57 @@ sub verify_multipath_configuration {
     record_info('configuration', 'Verify multipath configuration with multipath tool');
     my $conf_output = script_output("multipath -t");
     while (my ($k, $v) = each(%{$test_data->{attributes}})) {
-        assert_matches(qr/$k\s$v/, $conf_output, "Multipath attribute '$k $v' not found");
+        assert_matches(qr/$k\s"?$v"?/, $conf_output, "Multipath attribute '$k $v' not found");
     }
 }
 
 sub verify_multipath_topology {
-    my ($test_data) = shift;
+    my (%args) = @_;
 
-    my $wwid = $test_data->{WWID};
-    my $dm_N = $test_data->{device_map}->{hd0};
+    my $wwid      = $args{wwid};
+    my $test_data = $args{test_data};
 
     record_info('topology', 'Verify multipath topology');
-    my $topology_output = script_output("multipath -l");
-
-    # Check infomation header
-    assert_matches(qr/$wwid $dm_N/, $topology_output,
-        "Topology information does not show header info with WWID dm-N");
-    # Check paths
-    while (my ($dev, $values) = each(%{$test_data->{topology}})) {
-        # Check policy
-        assert_matches(qr/prio=$values->{prio} status=$values->{status}/, $topology_output,
-            "Policy unexpected for multipath path to $dev");
-        # Check device
-        assert_matches(qr/$dev.*active.*running/, $topology_output,
-            "Device '$dev' should be active and running");
-    }
-}
-
-sub verify_paths_list {
-    my ($test_data) = shift;
-    my $wwid = $test_data->{WWID};
-
-    record_info('paths', 'Verify multipath paths');
-    while (my ($dev, $values) = each(%{$test_data->{paths_list}})) {
-        validate_script_output("multipath -d -v3 | grep ^$wwid",
-            qr/$wwid.*$dev.*$values->{vendor_product_revision}/);
+    my $topology_output = script_output("multipath -ll");
+
+    # Check general topology info
+    my $topology    = $test_data->{topology};
+    my $ven_pro_rev = $topology->{vendor_product_revision};
+    assert_matches(qr/$wwid dm-0 $ven_pro_rev/, $topology_output,
+        'General topology info are not displayed properly');
+
+    # Check specific topology info
+    my $features  = $topology->{features};
+    my $hwhandler = $topology->{hwhandler};
+    my $wp        = $topology->{wp};
+    assert_matches(qr/size=.* features='$features' hwhandler='$hwhandler' wp=$wp/,
```

Comment:
could you make the size a bit more specific. i know it doesnt matter but it would be better to have an accurate regex here

Reply:
in general i dont have problem being too restrict with the regex. However the regex is a way to understand you data. it help in the debugging and in the design. here the size will match with everything. it is worth some caring, to represent your data. Nothing very pretentious

Reply:
It is not about to be strict, a good regex is better than a bad one, but I just chose not to check the size because I think it is not the point of this test. Parse fully the output blindly doesn't make any sense I think here I check too much things which basically are tested upstream (but more of that offline I can explain you).